File: services/agent_eval/presentation/api/routers/tasks.py
from fastapi import APIRouter
from core.use_cases.registry import TaskRegistry
from ...schemas.run_schemas import RunSummaryResponse

router = APIRouter(prefix="/tasks", tags=["tasks"])
registry = TaskRegistry()

@router.get("/")
async def list_tasks():
    tasks = registry.list_tasks()
    return [task.model_dump() for task in tasks]

@router.get("/{task_id}/history")
async def task_history(task_id: str):
    return []

File: apps/dashboard/lib/api/tasks.ts
import { apiClient } from "./client";
import { Task } from "../types/task";

export const listTasks = () => 
    apiClient.get<Task[]>("/api/tasks");

export const getTaskHistory = (taskId: string) => 
    apiClient.get<any[]>(`/api/tasks/${taskId}/history`);
